js/app.js

```diff
@@ -0,0 +1,149 @@
+
+const buttonEl = document.getElementById('requestResourceButton');
+
+const contentDivEl = document.getElementById('contentContainer');
+
+buttonEl.addEventListener('click', getData);
+
+function createXHR(method, url, cb){
+  var newRequest = new XMLHttpRequest ();
+
+  newRequest.addEventListener('load', cb);
+  newRequest.addEventListener('error', displayError);
+  newRequest.open(method, url);
+  newRequest.send();
+}
+
+function displayError(){
+  contentDivEl.innerHTML = 'In galaxy far, far away, something went terribly wrong.';
```

Comment:
I would recommend adding something from the error itself so that the user might get a better understanding of what's gone wrong. The example in the README shows a not found for items that doesn't exist.